core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -118,6 +128,41 @@ protected void initChannel(Channel ch) throws Exception {
                                                       .get(ChannelOption.CONNECT_TIMEOUT_MILLIS);
     }
 
+    private void configureProxy(Channel ch, ProxyConfig proxyConfig, SslContext sslCtx) {
+        if (proxyConfig instanceof DisabledProxyConfig) {
```

Comment:
Or an enum, like ProxyType? E.g. ProxyType.SOCKS5

Reply:
Ah that's better. 😄

Reply:
So the reason I removed enum `ProxyType` previously was basically because I didn't like how when the end-user types `Proxy...`, intellij autocomplete would show `ProxyType`, but it doesn't really serve any purpose to the end-user.

For us, I definitely agree the `switch case` statement looks a lot cleaner 😄, so i guess this is a trade-off.

We can 1) keep as-is 2) revert back (I intentionally created a separate commit) 3)  move all proxy related classes to `com.linecorp.armeria.client` and keep `ProxyType` package private.

Let me know what you guys think 🙇

Reply:
SGTM :-)

Reply:
For me, I like the current code because I think it's much readable. 😉
Also, I don't think we don't have to move the package because:
- It doesn't harm to show one more class which starts with `Proxy...` because the IntelliJ shows them(there are so many classes already but we can know which one is from Armeria) with the package and I think the name, which is `ProxyType`, is clear to understand.